pymc3/distributions/dist_math.py

```diff
@@ -381,18 +382,17 @@ def perform(self, node, inputs, output_storage):
         x, = inputs
         output_storage[0][0] = np.asarray(self.spline(x))
 
-class DifferentiableSplineWrapper (SplineWrapper):
-    """
-    Creates a theano operation with defined gradient from
-    scipy.interpolate.UnivariateSpline
-    """
-
-    def __init__(self, spline):
-        super(DifferentiableSplineWrapper, self).__init__(spline)
-        self.spline_grad = SplineWrapper(spline.derivative())
-        self.__props__ += ('spline_grad',)
-
     def grad(self, inputs, grads):
         x, = inputs
         x_grad, = grads
-        return [x_grad * self.spline_grad(x)]
+
+        if not hasattr(self, 'grad_op'):
+            try:
+                self.grad_op = SplineWrapper(self.spline.derivative())
```

Comment:
I think we don't want to add `grad_op` to `__props__` because it would break `__hash__()` and `__eq__()`. For example, this code works now as expected, but wouldn't work if `grad_op` is added to `__props__`:
```python
s = InterpolatedUnivariateSpline(...)
op1 = SplineWrapper(s)
op2 = SplineWrapper(s)
op2.grad(...) # calculate gradient
assert op1 == op2
```
One would expect that `op1` be equal to `op2` because they wrap the same spline, but if `spline_op` is a property, it won't be the case.